mlrun/api/crud/model_endpoints.py

```diff
@@ -1034,45 +1182,254 @@ def build_kv_cursor_filter_expression(
+        }
+        return attributes
 
     @staticmethod
-    def get_access_key(auth_info: mlrun.api.schemas.AuthInfo):
-        access_key = auth_info.data_session
-        if not access_key:
-            raise mlrun.errors.MLRunBadRequestError("Data session is missing")
-        return access_key
+    def _json_loads_if_not_none(field: typing.Any) -> typing.Any:
+        return json.loads(field) if field is not None else None
 
     @staticmethod
-    def _get_batching_interval_param(intervals_list):
-        """Converting each value in the intervals list into a float number. None
-        Values will be converted into 0.0.
+    def get_endpoint_features(
+        feature_names: typing.List[str],
+        feature_stats: dict = None,
+        current_stats: dict = None,
+    ) -> typing.List[mlrun.api.schemas.Features]:
+        """
+        Getting a new list of features that exist in feature_names along with their expected (feature_stats) and
+        actual (current_stats) stats. The expected stats were calculated during the creation of the model endpoint,
+        usually based on the data from the Model Artifact. The actual stats are based on the results from the latest
+        model monitoring batch job.
+
+        param feature_names: List of feature names.
+        param feature_stats: Dictionary of feature stats that were stored during the creation of the model endpoint
+                             object.
+        param current_stats: Dictionary of the latest stats that were stored during the last run of the model monitoring
+                             batch job.
+
+        return: List of feature objects. Each feature has a name, weight, expected values, and actual values. More info
+                can be found under mlrun.api.schemas.Features.
+        """
 
-        example::
-        Applying the function on a scheduling policy that is based on every hour exactly
-        _get_batching_interval_param(intervals_list=[0, '*/1', None])
-        The result will be: (0.0, 1.0, 0.0)
+        # Initialize feature and current stats dictionaries
+        safe_feature_stats = feature_stats or {}
+        safe_current_stats = current_stats or {}
 
+        # Create feature object and add it to a general features list
+        features = []
+        for name in feature_names:
+            if feature_stats is not None and name not in feature_stats:
+                logger.warn("Feature missing from 'feature_stats'", name=name)
+            if current_stats is not None and name not in current_stats:
+                logger.warn("Feature missing from 'current_stats'", name=name)
+            f = mlrun.api.schemas.Features.new(
+                name, safe_feature_stats.get(name), safe_current_stats.get(name)
+            )
+            features.append(f)
+        return features
+
+    def get_endpoint_metrics(
+        self,
+        endpoint_id: str,
+        metrics: typing.List[str],
+        start: str = "now-1h",
```

Comment:
where that convention of `now-1h` is defined?  please reference to a class and add docs explaining it

Reply:
fixed:
```
:param start:                    The start time of the metrics. Can be represented by a string containing an RFC 3339
                                 time, a Unix timestamp in milliseconds, a relative time (`'now'` or
                                 `'now-[0-9]+[mhd]'`, where `m` = minutes, `h` = hours, and `'d'` =
                                 days), or 0 for the earliest time.
```